src/Tizen.NUI/src/public/Common/PropertyMap.cs

```diff
@@ -179,19 +229,72 @@ public PropertyMap Add(KeyValue keyValue)
             {
                 throw new global::System.ArgumentNullException(nameof(keyValue));
             }
-            if (keyValue.KeyInt != null)
+            if (keyValue.IntegerKey != null)
             {
-                Interop.PropertyMap.Add(SwigCPtr, (int)keyValue.KeyInt, PropertyValue.getCPtr(keyValue.TrueValue));
+                using (PropertyKey pKey = new PropertyKey((int)keyValue.IntegerKey))
+                {
+                    SetValue(pKey, keyValue.PropertyValue);
+                }
             }
-            else if (keyValue.KeyString != null)
+            else if (keyValue.StringKey != null)
             {
-                Interop.PropertyMap.Add(SwigCPtr, keyValue.KeyString, PropertyValue.getCPtr(keyValue.TrueValue));
+                using (PropertyKey pKey = new PropertyKey(keyValue.StringKey))
+                {
+                    SetValue(pKey, keyValue.PropertyValue);
+                }
             }
-            if (NDalicPINVOKE.SWIGPendingException.Pending) throw NDalicPINVOKE.SWIGPendingException.Retrieve();
 
             return this;
         }
 
+        /// <summary>
+        /// Removes the element by the specified key.
+        /// </summary>
+        /// <param name="key">The index key to find.</param>
+        /// <returns>True if the element is removed, false otherwise.</returns>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public bool Remove(PropertyKey key)
```

Comment:
Use exception for the consistency
```suggestion
        public void Remove(PropertyKey key)
```

Reply:
Done

Reply:
Dictionary.Remove returns false if key does not exist.
So PropertyMap follows this rule.